src/open_inwoner/configurations/admin.py

```diff
@@ -26,40 +23,14 @@
 from .models import CustomFontSet, SiteConfiguration, SiteConfigurationPage
 
 
-@admin.action(description=_("Delete selected websites"))
-def delete_selected(modeladmin, request, queryset):
-    """
-    Override `delete_selected` action to prevent accidental deletion of all websites
-    """
-    if queryset.count() == Site.objects.all().count():
-        messages.add_message(
-            request,
-            messages.WARNING,
-            _("You cannot delete all websites; at least one site must remain."),
-        )
-        return HttpResponseRedirect(reverse("admin:sites_site_changelist"))
-    return default_delete_selected(modeladmin, request, queryset)
-
-
 class CustomSiteAdmin(SiteAdmin):
-    """
-    Overrides `SiteAdmin` to prevent deletion of last website
-    """
-
     model = Site
-    actions = [delete_selected]
 
-    def delete_model(self, request, obj):
-        if self.model.objects.count() < 2:
-            messages.add_message(
-                request, messages.WARNING, _("You cannot delete the last site.")
-            )
-            return redirect("admin:sites_site_changelist")
-        else:
-            super().delete_model(request, obj)
+    def has_add_permission(self, request):
+        return not Site.objects.exists()
```

Comment:
Double check, this doesn't need an extra `.all()`?


```suggestion
        return not Site.objects.all().exists()
```

Reply:
They're equivalent. Double checked: `.all()` returns a copy of the queryset, and `.all().exists()` is equivalent to calling `.exists()` on the manager itself. 

#one-obvious-way-to-do-it